cpp/src/IceGrid/ServerI.cpp

```diff
@@ -313,120 +313,108 @@ class ResetPropertiesCB : public IceUtil::Shared
-        assert(v.size());
-        string::size_type beg = 0;
-        string::size_type end;
-#ifdef _WIN32
-        vector<wchar_t> buf;
-        buf.resize(32767);
-        while((beg = v.find("%", beg)) != string::npos && beg < v.size() - 1)
+        string variable = v.substr(beg + 1, end - beg - 1);
+        DWORD ret = GetEnvironmentVariableW(Ice::stringToWstring(variable).c_str(), &buf[0],
+                                            static_cast<DWORD>(buf.size()));
+        string valstr = (ret > 0 && ret < buf.size()) ? Ice::wstringToString(&buf[0]) : string("");
+        v.replace(beg, end - beg + 1, valstr);
+        beg += valstr.size();
+        }
+#else
+    while((beg = v.find("$", beg)) != string::npos && beg < v.size() - 1)
+    {
+        string variable;
+        if(v[beg + 1] == '{')
         {
-            end = v.find("%", beg + 1);
+            end = v.find("}");
             if(end == string::npos)
             {
                 break;
             }
-            string variable = v.substr(beg + 1, end - beg - 1);
-            DWORD ret = GetEnvironmentVariableW(Ice::stringToWstring(variable).c_str(), &buf[0],
-                                                static_cast<DWORD>(buf.size()));
-            string valstr = (ret > 0 && ret < buf.size()) ? Ice::wstringToString(&buf[0]) : string("");
-            v.replace(beg, end - beg + 1, valstr);
-            beg += valstr.size();
+            variable = v.substr(beg + 2, end - beg - 2);
         }
-#else
-        while((beg = v.find("$", beg)) != string::npos && beg < v.size() - 1)
+        else
         {
-            string variable;
-            if(v[beg + 1] == '{')
+            end = beg + 1;
+            while((isalnum(static_cast<unsigned char>(v[end])) || v[end] == '_')  && end < v.size())
             {
-                end = v.find("}");
-                if(end == string::npos)
-                {
-                    break;
-                }
-                variable = v.substr(beg + 2, end - beg - 2);
-            }
-            else
-            {
-                end = beg + 1;
-                while((isalnum(static_cast<unsigned char>(v[end])) || v[end] == '_')  && end < v.size())
-                {
-                    ++end;
-                }
-                variable = v.substr(beg + 1, end - beg - 1);
-                --end;
+                ++end;
             }
-
-            char* val = getenv(variable.c_str());
-            string valstr = val ? string(val) : "";
-            v.replace(beg, end - beg + 1, valstr);
-            beg += valstr.size();
+            variable = v.substr(beg + 1, end - beg - 1);
+            --end;
         }
-#endif
-        return value.substr(0, assignment) + "=" + v;
+
+        char* val = getenv(variable.c_str());
+        string valstr = val ? string(val) : "";
+        v.replace(beg, end - beg + 1, valstr);
+        beg += valstr.size();
     }
+#endif
+    return value.substr(0, assignment) + "=" + v;
 
 };
```

Comment:
can you fix the extra line and semicolon, they were already there but seems unnecessary